confgenerator/logging_processors.go

```diff
@@ -62,6 +63,52 @@ func init() {
 	LoggingProcessorTypes.RegisterType(func() Component { return &LoggingProcessorParseJson{} })
 }
 
+// rewriteComplexCaptures translates disallowed capture group names into placeholders and a transformation to rename them back in the log record.
+func rewriteComplexCaptures(regex, tag string) (string, []fluentbit.Component) {
+	// Short-circuit regexes that don't have disallowed capture group names
+	disallowed := regexp.MustCompile(`\(\?P<(?:[A-Za-z0-9_]*[^A-Za-z0-9_>])+(?:[A-Za-z0-9_]*)>`)
+	if !disallowed.MatchString(regex) {
+		return regex, nil
+	}
+	// Maintain a list of rewritten capture group names
+	var rewrites []string
+	captureGroup := regexp.MustCompile(`\(\?P<((?:[^>\\]|\\.)*)>`)
+	// Can't use ReplaceAllStringFunc, since it doesn't support replacing only captured values
+	groupIndexes := captureGroup.FindAllStringSubmatchIndex(regex, -1)
+	l := 0
+	var r []string
+	for _, i := range groupIndexes {
+		g := regex[i[0]:i[1]] // Full match
+		s := regex[i[2]:i[3]] // First capture group
+		r = append(r, regex[l:i[2]])
+		// Also replace any capture group whose name starts with "__"
+		if !disallowed.MatchString(g) && !strings.HasPrefix(s, "__") {
+			r = append(r, s)
+		} else {
+			rewrites = append(rewrites, s)
+			r = append(r, fmt.Sprintf("__%d", len(rewrites)))
+		}
+		l = i[3]
+	}
+	r = append(r, regex[l:])
+	// Reconstruct the regex
+	regex = strings.Join(r, "")
+	// Rename all captured fields
+	oc := make([][2]string, len(rewrites))
+	for i, g := range rewrites {
+		oc = append(oc, [2]string{"Rename", fmt.Sprintf("__%d %q", i+1, g)})
```

Comment:
This doesn't allow users to set complicated fields they might want to set such as `httpRequest.status`. I think you need to use `modify_fields` here to perform the rename operation so that it can parse all the field names correctly.

Reply:
Right, it doesn't, but this wasn't the intended goal. This was supposed to match the `parse_json` functionality. As I understand it, `parse_json` would not transform `{"httpRequest.status": 200}` into `LogEntry(httpRequest={status=200})` either… For that, you'd need something like the `modify_fields` processor you're adding in #474.

Reply:
We should not match `parse_json`'s behavior here. It intentionally has a different input shape.

Reply:
What about:

1.  `(?<logging.googleapis.com/labels/hello>...)`

2.  `(?<"logging.googleapis.com/labels".hello>...)`

The second form is basically the filter syntax, except it uses a non-LogEntry structure which would be confusing...

If `severity` is the only one that doesn't need nesting, and `severity` is the one causing the most trouble for users, what if we just forget about nested fields? Seems like it would still add value even if `severity` is the only one that really works with the syntax.